test/views/components/H1Spec.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.select.Elements
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.Application
+import play.api.i18n.Messages
+import utils.SpecBase
+import views.html.components.h1
+
+class H1Spec extends SpecBase {
+
+  "h1 component" should {
+
+    "display correct contents" when {
+
+      "only message key has been provided" in new Setup {
+        h1Component.text() mustBe message(msgKey)
+
+        val h1: Elements = h1Component.select("h1")
+        h1.size() mustBe 1
+        h1.first().hasAttr("class") mustBe true
+        h1.first().classNames() must contain(defaultClass)
+        h1.first().hasAttr("id") mustBe false
+      }
+
+      "class has been provided along with message key" in new Setup {
+        h1ComponentWithClass.text() mustBe message(msgKey)
+
+        val h1: Elements = h1ComponentWithClass.select("h1")
+        h1.first().classNames() must contain(classes)
+      }
+
+      "id has been provided along with message key" in new Setup {
+        h1ComponentWithId.text() mustBe message(msgKey)
+
+        val h1: Elements = h1ComponentWithId.select("h1")
+        h1.first().attr("id") mustBe id
+      }
+
+      "classes and id have been provided along with message key" in new Setup {
+        h1ComponentWithClassAndId.text() mustBe message(msgKey)
+
+        val h1: Elements = h1ComponentWithClassAndId.select("h1")
```

Comment:
Better to give a line break after this line to differentiate between result and assertions

Reply:
This has been addressed in the commit [73dfe78](https://github.com/hmrc/customs-financials-documents-frontend/pull/101/commits/73dfe781532fd8bddb0967c9eeb15974440beaf7)